docs/docs/api-ref.md

```diff
@@ -0,0 +1,261 @@
+---
+title: API reference
+layout: docs
+---
+
+<div class="alert alert-info">
+    <b>Important:</b> More coming soon! This section is under active construction and might not reflect all the available functionality of the TSC library.
+    Until this reference is completed, we have noted the source files in the TSC library where you can get more information for individual endpoints.
+</div>
+
+* TOC
+{:toc}
+
+## Authentication
+
+Source files: server/endpoint/auth_endpoint.py, models/tableau_auth.py
+
+### Sign In
+
+Signs you in to Tableau Server.
+
+```py
+Auth.sign_in(authentication_object)
+```
+
+### Sign Out
+
+Signs you out of Tableau Server.
+
+```py
+Auth.sign_out()
+```
+
+## Sites
+
+Source files: server/endpoint/sites_endpoint.py, models/site_item.py
+
+### Create Site
+
+Creates a new site for the given site item object.
+
+```py
+Sites.create(new_site_item)
+```
+
+Example:
+
+```py
+new_site = TSC.SiteItem(name='Tableau', content_url='tableau', admin_mode=TSC.SiteItem.AdminMode.ContentAndUsers, user_quota=15, storage_quota=1000, disable_subscriptions=True)
+self.server.sites.create(new_site)
+```
+
+### Get Site by ID
+
+Gets the site with the given ID.
+
+```py
+Sites.get_by_id(id)
+```
+
+### Get Sites
+
+Gets all the sites on the server.
+
+```py
+Sites.get()
+```
+
+### Update Site
+
+Modifies a site. The site item object must include the site ID and overrides all other settings.
+
+```py
+Sites.update(site_item_object)
+```
+
+### Delete Site
+
+Deletes the site with the given ID.
+
+```py
+Sites.delete(id)
+```
+
+## Projects
+
+Source files: server/endpoint/projects_endpoint.py
+
+### Create Project
+
+Creates a project for the given project item object.
+
+```py
+Projects.create(project_item_object)
+```
+
+Example:
+
+```py
+new_project = TSC.ProjectItem(name='Test Project', description='Project created for testing')
```

Comment:
I pulled in most of the examples from the test suite in the interest of time--I expect I'll be rewriting more cogent examples afterward. :(